tests/http/server/_base.py

```diff
@@ -98,3 +111,413 @@ def _test_disconnect(
                 self.assertEqual(code, expected_code)
                 self.assertEqual(request.code, expected_code)
                 self.assertEqual(body, expected_body)
+
+    @logcontext_clean
+    def _test_cancellation_at_every_await(
+        self,
+        reactor: ThreadedMemoryReactorClock,
+        make_request: Callable[[], FakeChannel],
```

Comment:
rather than passing this in, can we just take a `*args, **kwargs` to pass through to `make_request` ?

Reply:
I was thinking that we might want to test federation requests in future, which have their own `make_signed_federation_request` function
https://github.com/matrix-org/synapse/blob/9d8e380d2e8267129de921b9b926257c36417cd2/tests/unittest.py#L795

though there is nothing stopping us from defining a `make_signed_federation_request_with_cancellation_test` variant.